project-management-openproject/project-management-openproject-ui/src/main/resources/ProjectManagement/OpenProjectSourceJson.xml

```diff
@@ -0,0 +1,103 @@
+  <web>ProjectManagement</web>
+  <name>OpenProjectSourceJson</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>ProjectManagement.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>OpenProjectSourceJson</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity wiki="false"}}
+#set ($configClassName = 'ProjectManagement.OpenProjectConnectionClass')
+#set ($query = "from doc.object($configClassName) as connection")
+#set ($connectionsDocs = $services.query.xwql($query).execute())
+#set ($connObjects = [])
+#foreach($connectionDoc in $connectionsDocs)
+    #set ($document = $xwiki.getDocument($connectionDoc))
+    #set ($connection = $document.getObject($configClassName))
+    #set ($discard = $connObjects.add($connection))
+#end
+#set ($results = $connObjects)
+{
+  "totalrows": $results.size(),
+  "matchingtags": {},
+  "tags" : [],
+  "returnedrows": $results.size(),
+  "offset": 0,
+  "sort": "$request.sort",
+  "dir": "$request.dir",
+  "reqNo": $numbertool.toNumber($request.reqNo).intValue(),
+  "rows" : [
+  #foreach($result in $results)
+  #set ($documentReference = $result.getDocumentReference())
+  #set ($connectionName = "&lt;span class='text-muted'&gt;$!result.get('connectionName')&lt;/span&gt;")
+  #set ($serverURL = "&lt;a href='http://$!result.get('serverURL')' target='_blank'&gt;$!result.get('serverURL')&lt;/a&gt;")
+  #set ($clientId = "&lt;span class='text-primary'&gt;$!result.get('clientId')&lt;/span&gt;")
+  #set ($clientSecret = "&lt;span class='text-primary h1'&gt;••••••••&lt;/span&gt;")
+  #set ($conn = $jsontool.serialize($result.get("connectionName")))
+  #set ($sURL = $jsontool.serialize($result.get("serverURL")))
+  #set ($cId = $jsontool.serialize($result.get("clientId")))
+  #set ($cS = $jsontool.serialize($result.get("clientSecret")))
+  #set ($actionsHTML = "
+  &lt;button
+    class='edit-button btn btn-link'
+    data-connectionName=$conn
+    data-serverURL=$sURL
+    data-clientId=$cId
+    data-clientSecret=$cS
+    data-documentReference=$documentReference
+  &gt;
+    &lt;i class='fa fa-pencil text-muted'&gt;&lt;/i&gt;
+  &lt;/button&gt;
+  &lt;button
+    class='delete-button btn btn-link'
+    data-connectionName=$conn
+    data-document-reference=$documentReference
+  &gt;
+    &lt;i class='fa fa-trash text-danger'&gt;&lt;/i&gt;
+  &lt;/button&gt;
+")
+
+  #set ($actionsJSON = $jsontool.serialize($actionsHTML))
+  #if($foreach.count &gt; 1),#end
+  {
+    "connectionName" : $jsontool.serialize($connectionName),
+    "serverURL" : $jsontool.serialize($serverURL),
+    "clientId" : $jsontool.serialize($clientId),
+    "clientSecret" : $jsontool.serialize($clientSecret),
+    "actions" : $actionsJSON
+  }
+#end
+  ]
+}
```

Comment:
You should use the `#jsonReponse` macro.
Usage: https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-ui/src/main/resources/CKEditor/MacroService.xml#L279

Definition: https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm#L2918-L2919